### ISSUE TITLE
[Bug] 

### ISSUE THREAD

**Issue by Wuzhongze** [open-mmlab/mmdetection#9302]
### Prerequisite

- [X] I have searched [Issues](https://github.com/open-mmlab/mmdetection/issues) and [Discussions](https://github.com/open-mmlab/mmdetection/discussions) but cannot get the expected help.
- [X] I have read the [FAQ documentation](https://mmdetection.readthedocs.io/en/latest/faq.html) but cannot get the expected help.
- [X] The bug has not been fixed in the [latest version (master)](https://github.com/open-mmlab/mmdetection) or [latest version (3.x)](https://github.com/open-mmlab/mmdetection/tree/dev-3.x).

### Task

I have modified the scripts/configs, or I'm working on my own tasks/models/datasets.

### Branch

3.x branch https://github.com/open-mmlab/mmdetection/tree/3.x

### Environment

absl-py==1.0.0
addict==2.4.0
anyio==3.5.0
argon2-cffi==21.3.0
argon2-cffi-bindings==21.2.0
asttokens==2.0.5
attrs==21.4.0
Babel==2.10.1
backcall==0.2.0
beautifulsoup4==4.11.1
bleach==5.0.0
cachetools==5.0.0
certifi==2021.10.8
cffi==1.15.0
charset-normalizer==2.0.12
click==8.1.3
colorama==0.4.6
commonmark==0.9.1
cycler==0.11.0
debugpy==1.6.0
decorator==5.1.1
defusedxml==0.7.1
entrypoints==0.4
executing==0.8.3
fastjsonschema==2.15.3
fonttools==4.33.3
google-auth==2.6.6
google-auth-oauthlib==0.4.6
grpcio==1.45.0
idna==3.3
importlib-metadata==4.11.3
importlib-resources==5.7.1
ipykernel==6.13.0
ipython==8.2.0
ipython-genutils==0.2.0
jedi==0.18.1
Jinja2==3.1.1
joblib==1.1.0
json5==0.9.6
jsonschema==4.4.0
jupyter-client==7.3.0
jupyter-core==4.10.0
jupyter-server==1.17.0
jupyterlab==3.3.4
jupyterlab-language-pack-zh-CN==3.3.post2
jupyterlab-pygments==0.2.2
jupyterlab-server==2.13.0
keras==2.8.0
kiwisolver==1.4.2
Markdown==3.3.6
MarkupSafe==2.1.1
matplotlib==3.5.1
matplotlib-inline==0.1.3
mistune==0.8.4
mmcv==2.0.0rc2
mmengine==0.3.1
model-index==0.1.11
nbclassic==0.3.7
nbclient==0.6.0
nbconvert==6.5.0
nbformat==5.3.0
nest-asyncio==1.5.5
notebook==6.4.11
notebook-shim==0.1.0
numpy==1.22.3
nvidia-ml-py==11.450.51
nvitop==0.5.5
oauthlib==3.2.0
opencv-contrib-python==4.5.5.64
opencv-python-headless==4.6.0.66
openmim==0.3.3
ordered-set==4.1.0
packaging==21.3
pandas==1.4.2
pandocfilters==1.5.0
parso==0.8.3
pexpect==4.8.0
pickleshare==0.7.5
Pillow==9.1.0
prometheus-client==0.14.1
prompt-toolkit==3.0.29
protobuf==3.20.1
psutil==5.9.0
ptyprocess==0.7.0
pure-eval==0.2.2
py3nvml==0.2.7
pyasn1==0.4.8
pyasn1-modules==0.2.8
pycocotools==2.0.6
pycparser==2.21
Pygments==2.12.0
pyparsing==3.0.8
pyrsistent==0.18.1
python-dateutil==2.8.2
pytz==2022.1
PyYAML==6.0
pyzmq==22.3.0
requests==2.27.1
requests-oauthlib==1.3.1
rich==12.6.0
rsa==4.8
scikit-learn==1.0.2
scipy==1.8.0
Send2Trash==1.8.0
six==1.16.0
sklearn==0.0
sniffio==1.2.0
soupsieve==2.3.2.post1
stack-data==0.2.0
tabulate==0.9.0
tensorboard==2.8.0
tensorboard-data-server==0.6.1
tensorboard-plugin-wit==1.8.1
termcolor==1.1.0
terminado==0.13.3
terminaltables==3.1.10
threadpoolctl==3.1.0
tinycss2==1.1.1
torch @ file:///root/torch-1.10.2%252Bcu113-cp38-cp38-linux_x86_64.whl
torchaudio==0.10.2+cu113
torchvision==0.11.3+cu113
tornado==6.1
traitlets==5.1.1
typing_extensions==4.2.0
urllib3==1.26.9
wcwidth==0.2.5
webencodings==0.5.1
websocket-client==1.3.2
Werkzeug==2.1.1
xgboost==1.6.0
xmltodict==0.12.0
yapf==0.32.0
zipp==3.8.0

### Reproduces the problem - code sample

/mmdet/evaluation/metrics/coco_metric.py  -437L            
# evaluate proposal, bbox and segm
            iou_type = 'bbox' if metric == 'proposal' else metric
            if metric not in result_files:
                raise KeyError(f'{metric} is not in results')
            try:
                predictions = load(result_files[metric]) 
                if iou_type == 'segm':
                    # Refer to https://github.com/cocodataset/cocoapi/blob/master/PythonAPI/pycocotools/coco.py#L331  # noqa
                    # When evaluating mask AP, if the results contain bbox,
                    # cocoapi will use the box area instead of the mask area
                    # for calculating the instance area. Though the overall AP
                    # is not affected, this leads to different
                    # small/medium/large mask AP results.
                    for x in predictions:
                        x.pop('bbox')
                coco_dt = self._coco_api.loadRes(predictions)

            except IndexError:
                logger.error(
                    'The testing results of the whole dataset is empty.')
                break

### Reproduces the problem - command or script

python /mmdetection/configs/soft_teacher/nodule-soft-teacher_faster-rcnn_r50-caffe_fpn_180k_semi-0.1-coco.py

### Reproduces the problem - error message

mmengine - ERROR - /hy-tmp/mmdetection/mmdet/evaluation/metrics/coco_metric.py - compute_metrics - 437 - The testing results of the whole dataset is empty.

**Comment by BIGWangYuDong** [comment]
Seems your model did not detect any boxes. Hence, please check whether your code(dataset, training setting) is normal, and also please check whether loss is normal in your log file.

**Comment by Wuzhongze** [comment]
> Seems your model did not detect any boxes. Hence, please check whether your code(dataset, training setting) is normal, and also please check whether loss is normal in your log file.
The following is the log information：
Accumulating evaluation results...
DONE (t=0.01s).
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.000
 Average Precision  (AP) @[ IoU=0.50      | area=   all | maxDets=1000 ] = 0.000
 Average Precision  (AP) @[ IoU=0.75      | area=   all | maxDets=1000 ] = 0.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= small | maxDets=1000 ] = 0.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area=medium | maxDets=1000 ] = 0.000
 Average Precision  (AP) @[ IoU=0.50:0.95 | area= large | maxDets=1000 ] = -1.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=100 ] = 0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=300 ] = 0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=   all | maxDets=1000 ] = 0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= small | maxDets=1000 ] = 0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area=medium | maxDets=1000 ] = 0.000
 Average Recall     (AR) @[ IoU=0.50:0.95 | area= large | maxDets=1000 ] = -1.000
11/15 11:43:11 - mmengine - INFO - bbox_mAP_copypaste: 0.000 0.000 0.000 0.000 0.000 -1.000
11/15 11:43:12 - mmengine - INFO - Evaluating bbox...
Loading and preparing results...
11/15 11:43:12 - mmengine - ERROR - /hy-tmp/mmdetection/mmdet/evaluation/metrics/coco_metric.py - compute_metrics - 437 - The testing results of the whole dataset is empty.
11/15 11:43:12 - mmengine - INFO - Iter(val) [10/10]  teacher/coco/bbox_mAP: 0.0000  teacher/coco/bbox_mAP_50: 0.0000  teacher/coco/bbox_mAP_75: 0.0000  teacher/coco/bbox_mAP_s: 0.0000  teacher/coco/bbox_mAP_m: 0.0000  teacher/coco/bbox_mAP_l: -1.0000  student/coco/bbox_mAP: 0.0000  student/coco/bbox_mAP_50: 0.0000  student/coco/bbox_mAP_75: 0.0000  student/coco/bbox_mAP_s: 0.0000  student/coco/bbox_mAP_m: 0.0000  student/coco/bbox_mAP_l: -1.0000
11/15 11:45:29 - mmengine - INFO - Iter(train) [3650/18000]  lr: 1.0000e-02  eta: 10:40:04  time: 2.7409  data_time: 0.0364  memory: 16686  loss: 0.0529  sup_loss_rpn_cls: 0.0024  sup_loss_rpn_bbox: 0.0027  sup_loss_cls: 0.0167  sup_acc: 98.6328  sup_loss_bbox: 0.0285  unsup_loss_rpn_cls: 0.0005  unsup_loss_rpn_bbox: 0.0000  unsup_loss_cls: 0.0021  unsup_acc: 100.0000  unsup_loss_bbox: 0.0000
11/15 11:47:41 - mmengine - INFO - Iter(train) [3700/18000]  lr: 1.0000e-02  eta: 10:37:43  time: 2.6385  data_time: 0.0362  memory: 17026  loss: 0.0562  sup_loss_rpn_cls: 0.0027  sup_loss_rpn_bbox: 0.0039  sup_loss_cls: 0.0177  sup_acc: 97.2656  sup_loss_bbox: 0.0297  unsup_loss_rpn_cls: 0.0008  unsup_loss_rpn_bbox: 0.0000  unsup_loss_cls: 0.0012  unsup_acc: 100.0000  unsup_loss_bbox: 0.0000
11/15 11:47:42 - mmengine - INFO - Evaluating bbox...

**Comment by ZwwWayne** [comment]
Seems that you are training a semi-supervised model, and it failed somehow and the model is not successfully trained.
